packages/knip/src/util/glob-core.ts

```diff
@@ -130,6 +131,17 @@ export const findAndParseGitignores = async (cwd: string) => {
 
   if (isFile('.git/info/exclude')) addFile('.git/info/exclude', cwd);
 
+  const globalExcludesFile = getGitConfigValue('core.excludesFile', undefined, '--global');
+  if (globalExcludesFile !== undefined && isFile(globalExcludesFile)) {
+    addFile(globalExcludesFile, cwd);
+  } else if (typeof process.env.XDG_CONFIG_HOME === 'string' && process.env.XDG_CONFIG_HOME.trim() !== '') {
+    const xdgIgnore = `${process.env.XDG_CONFIG_HOME}/git/ignore`;
+    if (isFile(xdgIgnore)) addFile(xdgIgnore, cwd);
```

Comment:
Following the logic of the docs here:

> Its default value is `$XDG_CONFIG_HOME/git/ignore`. If `$XDG_CONFIG_HOME` is either not set or empty, `$HOME/.config/git/ignore` is used instead.

Which I read to mean that `isFile(xdgIgnore) === false` means "give up", not "try `$HOME`".